src/style/style.js

```diff
@@ -1266,7 +1270,7 @@ class Style extends Evented {
 
     queryRenderedFeatures(queryGeometry: PointLike | [PointLike, PointLike], params: any, transform: Transform) {
         if (params && params.filter) {
-            this._validate(validateStyle.filter, 'queryRenderedFeatures.filter', params.filter, null, params);
+            this._validate(validateStyle.filter, 'queryRenderedFeatures.filter', params.filter, {layerType: 'fill'}, params);
```

Comment:
Why this line diff defaults to `{layerType: 'fill'}`?

Reply:
Oh because for validation purposes, every other filter still uses the old `filterSpec` object, which is the same as the new `fill` filterSpec object. This could also be defaulted inside the validation function so it doesn't have to be passed down.

Reply:
Yes it could be more appropriate to default inside validation, as long as we're not introducing any side effects by doing that. But it might seem worth of a comment mentioning that it's for legacy, this got me a bit confused and I couldn't find the reason why looking inside that function.

Reply:
done 👍🏽